src/lib/ngsi/EntityId.cpp

```diff
@@ -312,6 +306,11 @@ bool EntityId::equal(EntityId* eP)
     return true;
   }
 
+  if (eP->isTypePattern == eP->isTypePattern)
+  {
+    return true;
+  }
+
   return false;
```

Comment:
Yes, seems to be a clearer implementation.

(As discussed by skype, to be fixed by Ken)